test/src/unit-cppapi-string-dims.cc

```diff
@@ -109,4 +193,1575 @@ TEST_CASE(
 
   if (vfs.is_dir(array_name))
     vfs.remove_dir(array_name);
-}
\ No newline at end of file
+}
+
+TEST_CASE(
+    "C++ API: Test default string dimensions",
+    "[cppapi][string-dim][default]") {
+  const std::string array_name = "cpp_unit_array";
+  Context ctx;
+
+  std::vector<char> dim1 = {'a', 'b', 'b', 'c', 'd', 'e', 'e', 'f'};
+  write_array_1(ctx, array_name, dim1, false);
+
+  std::vector<std::string> expected_dim1, collect_results_dim1;
+  std::vector<int32_t> expected_dim2, collect_results_dim2;
+  std::vector<std::string> expected_dim3, collect_results_dim3;
+  std::vector<int32_t> expected_a1_data, collect_results_a1;
+
+  int which_option = -1;
+  SECTION("1") {  // ch7065 reported to have failed
```

Comment:
NIT: Any way we could give those a meaningful short name?

Reply:
What would be meaningful ("VariantN", "OptionN", "AlternateN", "StatementSubsetN", "???")?
What would be short and meaningful...?
Open to suggestions...

Reply:
Mostly a nit... Let's not hold merging on it if it's not easy to come up with something.